[Doc] Add document for XYZW accessors of vector (#1797)
taichi-dev/taichi: docs/vector.rst

    TODO: add descriptions about ``a(i, j)``

XYZW vector component accessors
+++++++++++++++++++++++++++++++

We also provide four handy accessors for the first four vector components:

.. attribute:: a.x

   Same as ``a[0]``.

.. attribute:: a.y

   Same as ``a[1]``.

.. attribute:: a.z

   Same as ``a[2]``.

.. attribute:: a.w

   Same as ``a[3]``.

.. note::

>    XYZW accessors can be used both read and write::

Reviewer: ```suggestion
   XYZW accessors can be used for both read and write:
```

Author: ```suggestion
   XYZW accessors can be used for both read and write::
```
The `::` is placed intentionally so that the code below could be interpreted as `code`.

>    XYZW accessors can be used for both read and write::

Reviewer 2: ```suggestion
   XYZW accessors can be used for both reading and writing::
```

      v = ti.Vector([2, 3, 4])
      print(v.x)  # 2
      print(v.y)  # 3
      print(v.z)  # 4
      v.y = 8
      print(v.y)  # 8

   XYZW accessors can be used in both Taichi-scope and Python-scope.

>    XYZW accessors doesn't work for ``ti.Matrix``.

Reviewer: ```suggestion
   XYZW accessors don't work for ``ti.Matrix``.
```

>    XYZW accessors don't work for ``ti.Matrix``.

Reviewer 2: Can this XYZW accessors be used in both `ti.Vector` and `ti.field`, except `ti.Matrix`? ;)

Author: Yes XYZW accessors can be used in both `ti.Vector` and `ti.Vector.field`.

>    For GLSL-alike shuffling accessors, consider use `taichi_glsl <https://taichi-glsl.readthedocs.io>`_::

Reviewer: ```suggestion
   For GLSL-alike shuffling accessors, consider using `taichi_glsl <https://taichi-glsl.readthedocs.io>`_::
```

   :parameter a: (ti.Vector or ti.Vector.field)
   :return: (scalar) return the dimensionality of vector ``a``

   E.g.,
   ::

        # Taichi-scope
        a = ti.Vector([1, 2, 3])
        a.n  # 3

   ::

        # Python-scope
        a = ti.Vector.field(3, dtype=ti.f32, shape=(4, 5))
        a.n  # 3

.. note::

>    When used as a global vector field, it will additionally contain all the

Reviewer 2: ```suggestion
   When this attribute is used in a global vector field, it will additionally contain all the
```
Maybe specifying the subject can make it more clear, HDYT?

Author: In fact the subject is `this class`, i.e. `ti.Vector`.

>    When used as global vector fields, it will additionally have all the
>    metadata as a scalar field would have, E.g.::

Reviewer: ```suggestion
   When used as a global vector field, it will additionally contain all the
   metadata that a scalar field would have, E.g.::
```
